File: driver_app/lib/models/driver.dart
class Driver {
  final String id;
  final String name;
  final String phone;
  final String? email;
  final String status;
  final String? licenseNumber;
  final Map<String, dynamic>? currentLocation;

  Driver({
    required this.id,
    required this.name,
    required this.phone,
    this.email,
    required this.status,
    this.licenseNumber,
    this.currentLocation,
  });

  factory Driver.fromJson(Map<String, dynamic> json) {
    return Driver(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
      email: json['email'],
      status: json['status'] ?? 'AVAILABLE',
      licenseNumber: json['license_number'],
      currentLocation: json['current_location'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'phone': phone,
      'email': email,
      'status': status,
      'license_number': licenseNumber,
      'current_location': currentLocation,
    };
  }
}

File: backend/app/models/driver.py
"""Driver models"""
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional
from datetime import datetime, timezone
import uuid
from .enums import DriverStatus


class Driver(BaseModel):
    model_config = ConfigDict(extra="ignore")
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    name: str
    email: str
    phone: str
    license_number: str
    status: DriverStatus = DriverStatus.AVAILABLE
    current_location: Optional[dict] = None
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))


class DriverCreate(BaseModel):
    name: str
    email: str
    phone: str
    license_number: str


class DriverStatusUpdate(BaseModel):
    status: DriverStatus
    reason: Optional[str] = None


class DriverOTPRequest(BaseModel):
    phone: str


class DriverOTPVerify(BaseModel):
    phone: str
    otp: str


class DriverLocationUpdate(BaseModel):
    latitude: float
    longitude: float
    accuracy: Optional[float] = None
    speed: Optional[float] = None
    heading: Optional[float] = None
    timestamp: Optional[datetime] = None


class DriverLocation(BaseModel):
    model_config = ConfigDict(extra="ignore")
    driver_id: str
    latitude: float
    longitude: float
    accuracy: Optional[float] = None
    speed: Optional[float] = None
    heading: Optional[float] = None
    timestamp: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    trip_id: Optional[str] = None
